### ISSUE TITLE
[Suggestion] View Only Sessions

### ISSUE THREAD

**Issue by Malik12tree** [JannisX11/blockbench#1742]
### Concept
As the title suggests, when someone creates a session, they have the option to set it to view only for other clients joining the session (maybe if client inputs an admin pass, they're now also an admin).

For security, only the view port is available as that for rendering is done on one random session admin each frame.

Perhaps clients can also request an action and an admin can accept, reject or ban this action to not show again.

### Why
For more secure and protected models when doing commissions.

**Comment by JannisX11** [comment]
Is your concern that clients (or generally session guests) may download the model while you are showing it? Preventing that may be difficult since Blockbench can be almost entirely modified by plugins and doesn't provide a platform that would be safe for a feature like that.
Just preventing edits from guests would be more realistic.

**Comment by Malik12tree** [comment]
> Is your concern that clients (or generally session guests) may download the model while you are showing it? Preventing that may be difficult since Blockbench can be almost entirely modified by plugins and doesn't provide a platform that would be safe for a feature like that.

*For security, only the view port is available as that for rendering is done on one random session admin each frame.*
Rendering is done on an admin's device. So you're now receiving image frames rather than elements.

> Just preventing edits from guests would be more realistic.

Great idea

**Comment by JannisX11** [comment]
> Rendering is done on an admin's device. So you're now receiving image frames rather than elements.
That's just Discord screen-sharing. I don't think we need to re-invent that.

**Comment by Malik12tree** [comment]
> That's just Discord screen-sharing. I don't think we need to re-invent that.

:p ! By rendering at an admins I meant in a virtual scene. By that view port movement is available.

**Comment by JannisX11** [comment]
Yeah I don't think that's viable with all the different variables that go into it, the latency, etc. But I'll keep it open for view-only.
